exporter/containerimage/opts.go

```diff
@@ -0,0 +1,109 @@
+package containerimage
+
+import (
+	"strconv"
+
+	cacheconfig "github.com/moby/buildkit/cache/config"
+	"github.com/moby/buildkit/util/compression"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	keyImageName        = "name"
+	keyLayerCompression = "compression"
+	keyCompressionLevel = "compression-level"
+	keyForceCompression = "force-compression"
+	keyOCITypes         = "oci-mediatypes"
+	keyBuildInfo        = "buildinfo"
+	keyBuildInfoAttrs   = "buildinfo-attrs"
+
+	// preferNondistLayersKey is an exporter option which can be used to mark a layer as non-distributable if the layer reference was
+	// already found to use a non-distributable media type.
+	// When this option is not set, the exporter will change the media type of the layer to a distributable one.
+	preferNondistLayersKey = "prefer-nondist-layers"
+)
+
+type ImageWriterOpts struct {
+	ImageName      string
+	RefCfg         cacheconfig.RefConfig
+	OCITypes       bool
+	BuildInfo      bool
+	BuildInfoAttrs bool
+}
+
+func (c *ImageWriterOpts) Parse(opt map[string]string) (map[string]string, error) {
```

Comment:
Think I missed this in first review but expected signature from `Parse` is `Parse(in) (out Type, .., error)`. If it is changing the state of the caller object it should be called `Load()` or `Initialize()`.

Reply:
Aha, good to know. A pure `Parse` method doesn't really work here, since we need a way of setting defaults before we extract options, so a `.Load`/`.Initialize` makes more sense.